libs/types-common/src/Data/Misc.hs

```diff
@@ -348,3 +351,12 @@ instance FromJSON PlainTextPassword where
 instance Arbitrary PlainTextPassword where
   -- TODO: why 6..1024? For tests we might want invalid passwords as well, e.g. 3 chars
   arbitrary = PlainTextPassword . fromRange <$> genRangeText @6 @1024 arbitrary
+
+-- | Usage:
+-- 1. Use this type in patterns to mark FUTUREWORKS.
+-- 2. Remove the label constructor -> all futureworks become compiler errors
+--
+-- Example:
+-- >>> let (FutureWork @'LegalholdPlusFederationNotImplemented -> _remoteUsers, localUsers)
+-- >>>      = partitionRemoteOrLocalIds domain qualifiedUids
+newtype FutureWork label payload = FutureWork payload
```

Comment:
If you plan to use the payload then uwrapping the newtype is a bit noisy
```hs
let FutureWork x' = FutureWork @'FeatNotImplemented x
```
If you also provide an unwrapper `FutureWork -> payload` and combine wrapper and unwrapper for convenience, this degenerates to

```
  let  futurework :: forall a b. b -> b
       futurework = id

  let x = futurework @'FeatNotImplemented (42 :: Int)
```
Having `FutureWork` type however has the advantage that you can pass futurework-values to functions in a typesafe way. So I'm in favor of `FutureWork` now after some thinking :)